library/core/src/main/java/com/google/android/exoplayer2/source/MaskingMediaSource.java

```diff
@@ -140,11 +140,6 @@ protected void onChildSourceInfoRefreshed(
     @Nullable MediaPeriodId idForMaskingPeriodPreparation = null;
     if (isPrepared) {
       timeline = timeline.cloneWithUpdatedTimeline(newTimeline);
-      if (unpreparedMaskingMediaPeriod != null) {
-        // Reset override in case the duration changed and we need to update our override.
-        setPreparePositionOverrideToUnpreparedMaskingPeriod(
-            unpreparedMaskingMediaPeriod.getPreparePositionOverrideUs());
-      }
```

Comment:
This seems remotely possible, perhaps if the timeline is growing when it is a newly started live playback where it has not reached its target window size.  @marcbaechinger what do you think?  My new test case still passes if we remove this change, however I'd suggest if we leave it that it is covered by a unit test case (right now it is not).